build/mergeallmodules.sh

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+
+#Assumes that ./x64 and ./x86 exist and contain the arch-specific copies of a DSC resource module
+
+zip -L >/dev/null 2>&1
+if [ $? != 0 ]; then
+        echo "zip must be installed"
+        exit 1
+fi
+
+unzip -Z >/dev/null 2>&1
+if [ $? != 0 ]; then
+        echo "unzip must be installed"
+        exit 1
+fi
+
+IntermediateDir="../intermediate"
+TargetDir="../target"
+DscModuleIntermediateDir="$IntermediateDir/merging_dsc_modules"
+DscModuleTargetDir="$TargetDir/dsc_modules"
+X64Dir="./x64"
+X86Dir="./x86"
+TestSigningDir=$1
+
+if [ "$TestSigningDir" = "" ]
+then
+	TestSigningDir="../test/config/testsigning"
+fi
+
+if [ -d "$TestSigningDir" ]
+then
+	gpg --list-secret-keys
+	if [ $? != 0 ]
+	then
+		echo "Failed to run the gpg command"
+		exit 1
+	fi
+
+	gpg --list-secret-keys | grep -q '2048R/8C3B51C6'
+	if [ $? != 0 ]
+	then
+		TestGPGKey="$TestSigningDir/gpg.asc"
+		if [ -f "$TestGPGKey" ]
+		then
+			echo "Import the test GPG key"
+			gpg --import $TestGPGKey
```

Comment:
Do you need sudo access for this, or is it imported into an account-specific keyring?